File: 226-invert-binary-tree/226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root == None:
            return root
        
        queue = deque()
        queue.append(root)
        
        while queue:
            levelSize = len(queue)
            
            for _ in range(levelSize):
                currNode = queue.popleft()
                
                currNode.left, currNode.right = currNode.right, currNode.left
                
                if currNode.left != None: queue.append(currNode.left)
                if currNode.right != None: queue.append(currNode.right)
        return root
        

File: 226-invert-binary-tree/226-invert-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        return solve(root);
    }
private:
    TreeNode* solve(TreeNode* root) {
        if(root == nullptr)
            return root;
        
        queue<TreeNode*> queue;
        queue.push(root);
        
        while(!queue.empty()) {
            int levelSize = queue.size();
            
            while(levelSize--) {
                auto currNode = queue.front();
                queue.pop();
                
                auto temp = currNode -> left;
                currNode -> left = currNode -> right;
                currNode -> right = temp;
                
                
                if(currNode -> left != nullptr) queue.push(currNode -> left);
                if(currNode -> right != nullptr) queue.push(currNode -> right);
            }
        }
        return root;
    }
};